### PR TITLE
Muon - Core Code; Load Widget Utilities

### PR DESCRIPTION
**Pull request by TWJubb**
Refs #23642 

### Summary

This PR creates a set of utility functions (with unit tests), for use with the load widget.

-----------

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
I think we should create a utils and a threading folder.

File: scripts/Muon/GUI/Common/muon_file_utils.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/muon_file_utils.py]: copyright is missing

File: scripts/Muon/GUI/Common/muon_file_utils.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/muon_file_utils.py]: I think we should have some sort of warning here

File: scripts/Muon/GUI/Common/muon_file_utils.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/muon_file_utils.py]: looking at the `_instrument_data_directory` method it calls `instrument_directory`, which is above. Why can we not pass in the instrument directory?

File: scripts/Muon/GUI/Common/muon_file_utils.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/muon_file_utils.py]: as far as I can tell up until (and including data) is just the output of `_instrument_data_directory`, so we don't we use that?

File: scripts/Muon/GUI/Common/muon_file_utils.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/muon_file_utils.py]: I think we should put this in a new utils folder within Common

File: scripts/Muon/GUI/Common/run_string_utils.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/run_string_utils.py]: also in utils folder

File: scripts/Muon/GUI/Common/thread_model.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/thread_model.py]: remove this file, as it is being covered in PR #23767

File: scripts/test/Muon/thread_model_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/thread_model_test.py]: same as above

File: scripts/test/Muon/utilities/run_string_utils_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/run_string_utils_test.py]: I think the file is getting a bit long, could we split these into two?

File: scripts/test/Muon/utilities/thread_model_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/thread_model_test.py]: covered in another PR, so we should remove it here

File: scripts/Muon/GUI/Common/muon_file_utils.py
Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/muon_file_utils.py]: I am pro simpler

File: scripts/Muon/GUI/Common/run_string_utils.py
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/run_string_utils.py]: can we have a comment explaining this some more

File: scripts/Muon/GUI/Common/muon_file_utils.py
Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/muon_file_utils.py]: Upon discussion it was decided to leave this as is for now.

**Comment by martyngigg** [comment]
Do we think we are nearly ready with this one? I think it's holding up a few of the reviewed PRs from being merged.

**Comment by AnthonyLim23** [comment]
the build failures look to be unrelated

**Comment by AnthonyLim23** [comment]
I would like to give it another look over once the other PRs have been merged

**Comment by martyngigg** [comment]
We've got some conflicts here now that #23645 has been merged.

File: scripts/test/Muon/utilities/muon_load_data_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_load_data_test.py]: can we check its not empty before clearing it?

File: scripts/test/Muon/utilities/muon_load_data_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_load_data_test.py]: same as above

File: scripts/test/Muon/utilities/muon_load_data_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_load_data_test.py]: same a above

File: scripts/test/Muon/utilities/muon_load_data_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_load_data_test.py]: same as above

File: scripts/test/Muon/utilities/muon_load_data_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_load_data_test.py]: the 3 above lines seem to be repeated a lot. Could we refactor them out to reduce repeated code?

File: scripts/test/Muon/utilities/run_string_utils_conversion_test.py
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/run_string_utils_conversion_test.py]: is context the best variable name?

**Review by AnthonyLim23 — APPROVED** [review]

**Pull request merged**
